static/src/js/api/announcements.js

```diff
@@ -0,0 +1,47 @@
+/* @file announcements.js
+ * @brief Provides the API wrappers for the announcements endpoints.
+ *
+ * @author Oscar Bezi (bezi@scottylabs)
+ */
+'use strict';
+
+var ajax = require('./ajax');
+
+var announcements = {};
+
+/* @brief The maximum length of an announcement. */
+announcements.maxLength = 500;
+
+/* @brief Returns a promise that resolves into a list of all announcements. */
+announcements.getAll = function () {
+  return ajax.get('/api/announcements');
+};
+
+/* @brief Makes a new announcement with the given data.
+ *
+ * A bit of a lie at the moment.  We disregard the client timestamp and use the
+ * server's, I'll get around to fixing it at some point (or not, the difference
+ * is tiny).
```

Comment:
Would we ever need to be able to specify the timestamp on the client side? Seems like it would be more consistent to only use the server's.